zetta_utils/mazepa_layer_processing/segmentation/adjust_affinities_with_masks.py

```diff
@@ -0,0 +1,109 @@
+from typing import Iterable
+
+import torch
+
+from zetta_utils import builder
+
+
+@builder.register("pass_all_data_if_any_masked")
+def pass_all_data_if_any_masked(
+    src: torch.Tensor,
+    masks: Iterable[torch.Tensor],
+) -> torch.Tensor:
+    """
+    If any mask is present, return the data, otherwise return zeros
+
+    :param src: input Tensor
+    :param masks: list of masks
+    """
+    is_masked = any(mask.sum() > 0 for mask in masks)
+    has_data = src.sum() > 0
```

Comment:
This doesn't account for cases where the data is signed - there should be check to see whether the `src` is indeed unsigned.